Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1062 @@
+                                        Write-Host $([Constants]::SingleDashLine)
+                                    }        
+                                }
+                                if(($null -ne $MinTLSVersion) -and ($apiError -eq $false))
+                                {
+                                    if($MinTLSVersion -ne $requiredMinTLSVersion)
+                                    {
+                                        $frontDoorEp.DomainName = $domainName
+                                        $frontDoorEp.FrontDoorName = $resourceName
+                                        $frontDoorEp.MinTLSVersion = $MinTLSVersion
+                                        $frontDoorEp.ResourceGroup = $resourceGroupName
+                                        $frontDoorEp.TypeOfCertificate = $TypeOfCertificate
+                                        $frontDoorEp.isMinTLSVersionSetOnCustomDomain = $false
+                                    
+                                        if($null -ne $frontDoorEp)
+                                        {
+                                            $NonCompliantFrontDoorEndpoints +=  $frontDoorEp | Select-Object @{N='ResourceGroupName';E={$frontDoorEp.ResourceGroup}},
+                                                                                        @{N='ResourceName';E={$frontDoorEp.FrontDoorName}},
+                                                                                        @{N='DomainName';E={$frontDoorEp.DomainName}},
+                                                                                        @{N='CertificateType';E={$frontDoorEp.TypeOfCertificate}},
+                                                                                        @{N='MinimumTlsVersion';E={$frontDoorEp.MinTLSVersion}},
+                                                                                        @{N='isMinTLSVersionSetOnCustomDomain';E={$frontDoorEp.isMinTLSVersionSetOnCustomDomain}}
+                                        }
+                                    }
+                                    else
+                                    {
+                                        $compliantFrontDoorEPs += $item
+                                    }
+                                }
+                                else
+                                {
+                                    Write-Host "Skipping this Front End Point..." -ForegroundColor $([Constants]::MessageType.Warning)
+                                    Write-Host $([Constants]::SingleDashLine)
+                                }
+                            }
+                    }
+                }
+                else
+                {
+                    if($frontDoorResource.MinimumTlsVersion -ne $requiredMinTLSVersion)
+                    {
+                        $NonCompliantFrontDoorEndpoints += $frontDoorResource
```

Comment:
We are adding resource at both level front door and endpoint in same collection, plz check if it will cause any impact while remediating resources.

Reply:
We are adding only into front door endpoint level. I have simplified the logic with comments.